src/vs/workbench/parts/debug/test/electron-browser/debugInlineDecorators.test.ts

```diff
@@ -0,0 +1,212 @@
+		assert.equal(decorator.renderOptions.dark.after.contentText, decorator.renderOptions.light.after.contentText);
+		assert.equal(decorator.renderOptions.dark.after.contentText, expectedDecoratorText);
+		assert.deepEqual(decorator.range, {
+			startLineNumber: lineNumber,
+			endLineNumber: lineNumber,
+			startColumn: testLine.length,
+			endColumn: testLine.length + 1
+		});
+	});
+
+	test('getDecorators returns correct decorator afterText', () => {
+		const lineContent = 'console.log(everything, part, part);'; // part shouldn't be duplicated
+		const lineNumber = 1;
+		const wordRangeMap = updateWordRangeMap(createStringDictionary<IRange[]>(), lineNumber, lineContent);
+		const decorators = inlineDecorators.getDecorators(testNameValueMap, wordRangeMap, [lineContent]);
+		const expectedDecoratorText = ' everything = {emmet: true, batman: true, legoUniverse: true}, part = "𝄞 ♪ ♫" ';
+		assert.equal(decorators[0].renderOptions.dark.after.contentText, expectedDecoratorText);
+	});
+
+	test('getEditorWordRangeMap ignores comments and long lines', () => {
+		const expectedWords = 'function, doit, everything, is, awesome, when, youre, part, of, a, team'.split(', ');
+		const editorModel = EditorModel.createFromString(`/** Copyright comment */\n  \n${testLine}\n// Test comment\n${createLongString()}\n`);
+		mockEditorModelLineTokens(editorModel);
+
+		const wordRangeMap = inlineDecorators.getEditorWordRangeMap(editorModel);
+		const words = Object.keys(wordRangeMap);
+		assert.deepEqual(words, expectedWords);
+	});
+});
+
+// Test helpers
+
+function createExpression(name: string, value: string): IExpression {
+	return {
+		name,
+		value,
+		getId: () => name,
+		hasChildren: false,
+		getChildren: null
+	};
+}
+
+function createLongString(): string {
+	let longStr = '';
+	for (let i = 0; i < 100; ++i) {
+		longStr += 'blah blah blah ';
+	}
+	return `"${longStr}"`;
+}
+
+// Simple word range creator that maches wordRegex throughout string
+function updateWordRangeMap(wordRangeMap: IStringDictionary<IRange[]>, lineNumber: number, lineContent: string): IStringDictionary<IRange[]> {
+	const wordRegexp = inlineDecorators.WORD_REGEXP;
+	wordRegexp.lastIndex = 0; // Reset matching
+
+	while (true) {
+		const wordMatch = wordRegexp.exec(lineContent);
+		if (!wordMatch) {
+			break;
+		}
+
+		const word = wordMatch[0];
+		const startOffset = wordMatch.index;
+		const endOffset = startOffset + word.length;
+
+		const range: IRange = {
+			startColumn: startOffset + 1,
+			endColumn: endOffset + 1,
+			startLineNumber: lineNumber,
+			endLineNumber: lineNumber
+		};
+
+		if (!wordRangeMap[word]) {
+			wordRangeMap[word] = [];
+		}
+
+		wordRangeMap[word].push(range);
+	}
+
+	return wordRangeMap;
+}
+
+interface MockToken {
+	tokenType: StandardTokenType;
+	startOffset: number;
+	endOffset: number;
+}
+
+// Simple tokenizer that separates comments from words
+function mockLineTokens(lineContent: string): LineTokens {
```

Comment:
I tried that. The issue is the tokens are whole line tokens since the `tokenizerSupport` is `null`  I tried using `model.setMode('javascript')` but that needs the javascript tokenizersupport to be registered and I couldn't fit all of the pieces to test that comments are ignored. So I wrote a simple stub that mocks `editor.getLineTokens()`.

Happy to change this if there is a way to plug in the javascript tokenizer.